Tests/GRPCTests/GRPCReflectionServiceTests/GRPCReflectionServiceTests.swift

```
@@ -0,0 +1,241 @@
+  }
+
+  func testListServices() async throws {
+    try self.setUpServerAndChannel()
+    let client = Reflection_ServerReflectionAsyncClient(channel: self.channel!)
+    let serviceReflectionInfo = client.makeServerReflectionInfoCall()
+    try await serviceReflectionInfo.requestStream.send(.with {
+      $0.host = "127.0.0.1"
+      $0.listServices = "services"
+    })
+    serviceReflectionInfo.requestStream.finish()
+    var iterator = serviceReflectionInfo.responseStream.makeAsyncIterator()
+    let message = try await iterator.next()
+    let receivedServices = message?.listServicesResponse.service.map { $0.name }
+    XCTAssertEqual(receivedServices?.count, 5)
+    for serviceName in receivedServices! {
+      XCTAssertTrue(self.servicesNames.contains(serviceName))
+    }
+  }
+
+  func testReflectionServiceData() throws {
+    self.createFileDescriptorProto()
+    do {
+      let registry = try ReflectionServiceData(fileDescriptorProtos: self.protos)
+      let registryFileDescriptorData = registry.getfileDescriptorData()
+      let services = registry.getServices()
+
+      for (fileName, protoData) in registryFileDescriptorData {
+        let serializedFiledescriptorData = protoData.getSerializedFileDescriptorProto()
+        let dependecies = protoData.getDependency()
+
+        let originalIndex = self.protos.firstIndex(where: { $0.name == fileName })
+        XCTAssertNotNil(originalIndex)
+
+        let originalProto = self.protos[originalIndex!]
+        XCTAssertEqual(originalProto.name, fileName)
+        XCTAssertEqual(try originalProto.serializedData(), serializedFiledescriptorData)
+        XCTAssertEqual(originalProto.dependency, dependecies)
+
+        self.protos.remove(at: originalIndex!)
+      }
+
+      for serviceName in services {
+        XCTAssert(self.servicesNames.contains(serviceName))
+        self.servicesNames.removeAll { $0 == serviceName }
+      }
+    } catch {
+      XCTFail(error.localizedDescription)
+    }
```

You don't need to do this: just let the test throw an error